components/script/dom/htmlimageelement.rs

```diff
@@ -375,6 +437,22 @@ impl HTMLImageElement {
         window.add_pending_reflow();
     }
 
+    fn process_image_response_for_environment_change(&self, image: ImageResponse) {
+        let element = self.upcast::<Element>();
+        let is_multipart_x_mixed_replace = HTMLImageElement::check_mutipart_x_mixed_replace(element);
+        if is_multipart_x_mixed_replace {
+            self.abort_request(State::Unavailable, ImageRequestPhase::Pending);
+        }
+
+        match (image, self.image_request.get()) {
```

Comment:
We should just be able to match on `image` here. We will need to modify the contents of `self.pending_request` in order to store the image data and metadata in it when we they are available (ImageResponse::Loaded, ImageResponse::PlaceholderLoaded, ImageResponse::MetadataLoaded). When we have a fully loaded image, we should invoke `finish_reacting_to_environment_changes`.

Reply:
In case of None, we still need to abort ?

Reply:
Yes.